lib/Agrammon/ResultCollector.pm6

```diff
@@ -3,7 +3,11 @@ use OO::Monitors;
 
 monitor Agrammon::ResultCollector {
     has %.results;
+    has %.validation-errors;
 
+    method add-validation-errors($simulation-name, $dataset-id, $errors --> Nil) {
+        %!validation-errors{$simulation-name}{$dataset-id} = $errors;
```

Comment:
It's a list of errors, so maybe `@errors` communicates better?

Reply:
done